distribution/build.gradle

```diff
@@ -139,11 +145,22 @@ tasks.register("buildTransportModules") {
   dependsOn "processTransportOutputs"
   outputs.dir "${transportOutputs}/modules"
 }
+tasks.register("buildExternalTestModules") {
+  dependsOn "processExternalTestOutputs"
+  outputs.dir "${externalTestOutputs}/modules"
+}
+
+Configuration moduleZip(Project module) {
+  Dependency dep = project.dependencies.project(path: module.path, configuration: 'zip')
+  Configuration config = project.configurations.detachedConfiguration(dep)
+  return config
+}
 
 void copyModule(Sync copyTask, Project module) {
+  Configuration moduleConfig = moduleZip(module)
   copyTask.configure {
-    dependsOn "${module.path}:bundlePlugin"
-    from({ zipTree(module.bundlePlugin.outputs.files.singleFile) }) {
+    dependsOn moduleConfig
+    from({ zipTree(moduleConfig.singleFile) }) {
```

Comment:
We should open an issue to use artifact transforms for this stuff as well so we don't have to keep doing this `zipTree` and `singleFile` nonsense. This is the same as the stuff @breskeby is working on to avoid a lot of unnecessary packing/unpacking in build processes.